### PR TITLE
MGMT-13612: Add confidential computing workflow

### PR DESCRIPTION
**Pull request by eranco74**
Added a periodic job with e2e-gcp-confidential workflow to 4.13 ci 
Add e2e-gcp-confidential optional job to installer-master

### CODE REVIEW

**Comment by eranco74** [comment]
pj-rehearse

File: ci-operator/step-registry/ipi/conf/gcp/confidential/OWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+approvers:
+- eranco74
+- omertuc
+- vrutkovs
+- jstuever
```
Comment (barbacbd) [inline ci-operator/step-registry/ipi/conf/gcp/confidential/OWNERS:5]: We can remove jstuever and replace with barbacbd

File: ci-operator/step-registry/ipi/conf/gcp/confidential/ipi-conf-gcp-confidential-ref.metadata.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+	"path": "ipi/conf/gcp/confidential/ipi-conf-gcp-confidential-ref.yaml",
+	"owners": {
+		"approvers": [
+			"eranco74",
+			"omertuc",
+			"vrutkovs",
+			"jstuever",
```
Comment (barbacbd) [inline ci-operator/step-registry/ipi/conf/gcp/confidential/ipi-conf-gcp-confidential-ref.metadata.json:8]: Can remove jstuever and replace with barbacbd

File: ci-operator/step-registry/openshift/e2e/gcp/confidential/OWNERS [hunk 3]
```diff
@@ -0,0 +1,6 @@
+approvers:
+- eranco74
+- omertuc
+- vrutkovs
+- jstuever
```
Comment (barbacbd) [inline ci-operator/step-registry/openshift/e2e/gcp/confidential/OWNERS:5]: replace jstuever with barbacbd

File: ci-operator/step-registry/openshift/e2e/gcp/confidential/openshift-e2e-gcp-confidential-workflow.metadata.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+	"path": "openshift/e2e/gcp/confidential/openshift-e2e-gcp-confidential-workflow.yaml",
+	"owners": {
+		"approvers": [
+			"eranco74",
+			"omertuc",
+			"vrutkovs",
+			"jstuever",
```
Comment (barbacbd) [inline ci-operator/step-registry/openshift/e2e/gcp/confidential/openshift-e2e-gcp-confidential-workflow.metadata.json:8]: replace jstuever with barbacbd

**Comment by barbacbd** [comment]
@eranco74  
Your new tests weren't executed. You will want to run those before we can pass this through.

**Comment by eranco74** [comment]
/pj-rehearse
/pj-rehearse pull-ci-openshift-installer-master-e2e-gcp-confidential
/pj-rehearse pull-ci-openshift-installer-master-e2e-gcp-secureboot

**Comment by eranco74** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.13-e2e-gcp-confidential

**Comment by eranco74** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.13-e2e-gcp-confidential

**Comment by eranco74** [comment]
time="2023-02-12T09:38:14Z" level=info msg="Cluster operator monitoring Progressing is True with RollOutInProgress: Rolling out the stack."
/pj-rehearse periodic-ci-openshift-release-master-ci-4.13-e2e-gcp-confidential

**Comment by eranco74** [comment]
```
openshift-monitoring                              prometheus-k8s-0                                                    0/6    Pending    0         42m
openshift-monitoring                              prometheus-k8s-1                                                    0/6    Pending    0         36m
```
```
40m        Warning  FailedMount               pod/prometheus-k8s-1                                         Unable to attach or mount volumes: unmounted volumes=[prometheus-data], unattached volumes=[secret-metrics-client-certs prometheus-k8s-rulefiles-0 secret-kube-etcd-client-certs secret-prometheus-k8s-proxy secret-prometheus-k8s-thanos-sidecar-tls kube-api-access-k6b89 secret-kube-rbac-proxy configmap-kubelet-serving-ca-bundle secret-prometheus-k8s-tls web-config configmap-serving-certs-ca-bundle config tls-assets configmap-metrics-client-ca config-out secret-grpc-tls prometheus-trusted-ca-bundle prometheus-data metrics-client-ca]: timed out waiting for the condition
```

**Comment by eranco74** [comment]
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/608
Seems that hypershift on GCP had the same issue few months ago
https://github.com/openshift/hypershift/pull/1874

**Comment by eranco74** [comment]
release-master-ci-4.13-e2e-gcp-confidential is failing due to:
https://issues.redhat.com/browse/OCPBUGS-7582

/pj-rehearse pull-ci-openshift-installer-release-4.13-e2e-gcp-secureboot

**Comment by eranco74** [comment]
/assign @dgoodwin 
/assign @patrickdillon
Note that I only changed the documentation of the secure boot workflow and the tests are failing due to failure to build the UPI image:
```
error occurred handling build upi-installer: the build upi-installer failed after 25m26s with reason DockerBuildFailed: Dockerfile build strategy has failed.
```
Other than that the periodic-ci-openshift-release-master-ci-4.14-e2e-gcp-confidential jo is failing due to a known issue: https://issues.redhat.com/browse/OCPBUGS-7582 that should get fixed in 4.14

**Comment by eranco74** [comment]
/retest

{ error occurred handling build upi-installer: the build upi-installer failed after 28m59s with reason DockerBuildFailed: Dockerfile build strategy has failed.}

**Pull request closed without merging**